### ISSUE TITLE
Page redirects lead to short url, not the canonical one

### ISSUE THREAD

**Issue by odinserj** [rtfd/readthedocs.org#1025]
Hello, I changed my documentation a bit, and these changes require me to make some redirects to respect the old links. So I've made the following page redirect in my admin panel:

```
users-guide/getting-started/installation.html
getting-started/installation.html
```

Redirect itself is working fine, but leads to another domain, non-canonical. For example, redirect descibed above leads from http://docs.hangfire.io/en/latest/users-guide/getting-started/installation.html to http://hangfire.readthedocs.org/en/latest/getting-started/installation.html. But I want docs to be redirected to the *docs.hangfire.io* domain instead of *hangfire.readthedocs.org*.

How to make redirect respect my canonical domain?

**Comment by lmjohns3** [comment]
I have uncovered a related issue: page redirects also do not appear to preserve the current version of the docs.

For example, my project (http://theanets.rtfd.org) has renamed a documentation file from ``creating.html`` to ``guide.html`` in the ``latest`` version, which at some point in the future will become the ``stable`` (default) version.

I added a page redirect for this (``creating.html --> guide.html``), but when I go to http://theanets.rtfd.org/en/latest/creating.html I get redirected to http://theanets.rtfd.org/en/stable/guide.html.

How can I get a page redirect to preserve the ``/en/latest/``? (It seems that preserving the ``/en/latest/`` is the intent, because when I added the redirect on the admin site, it listed the redirect as ``/$lang/$version/creating.html --> /$lang/$version/guide.html``.)
